File: Android/tuikaraoke/src/main/java/com/tencent/liteav/tuikaraoke/model/KaraokeMusicServiceObserver.java
package com.tencent.liteav.tuikaraoke.model;

import com.tencent.liteav.tuikaraoke.model.impl.base.KaraokeMusicInfo;
import com.tencent.liteav.tuikaraoke.model.impl.base.MusicPitchModel;

import java.util.List;

/**
 * 已点列表回调接口
 */
public interface KaraokeMusicServiceObserver {

    /**
     * 已点列表的更新
     *
     * @param musicInfoList 歌曲列表
     */
    void onMusicListChanged(List<KaraokeMusicInfo> musicInfoList);

    /**
     * 单句歌词的得分（百分制[0, 100]）
     *
     * @param currentScore 分数
     */
    void onMusicSingleScore(int currentScore);

    /**
     * 歌曲实时回调
     *
     * @param pitch       实时音高
     * @param timeStamp   歌词时间戳（播放进度）
     */
    void onMusicRealTimePitch(int pitch, float timeStamp);

    /**
     * 打分结果（百分制[0, 100]）
     *
     * @param totalScore 最终得分
     */
    void onMusicScoreFinished(int totalScore);

    /**
     * 打分准备就绪
     *
     * @param pitchModels 歌曲音高模型
     */
    void onMusicScorePrepared(List<MusicPitchModel> pitchModels);
}

File: iOS/Source/model/Impl/music/KaraokeMusicServiceObserver.swift
//
//  KaraokeMusicServiceObserver.swift
//  TUIKaraoke
//
//  Created by adams on 2023/4/3.
//

import UIKit

/**
 * 歌曲管理回调接口
 */
public protocol KaraokeMusicServiceObserver: AnyObject {
    /**
     * 歌曲列表更新回调
     * - parameter musicInfoList   已点歌曲列表数组
     */
    func onMusicListChanged(musicInfoList: [KaraokeMusicInfo])
    
    // MARK: - 打分相关
    
    /**
     * 打分准备就绪回调
     * - parameter pitchModelList 歌曲音高模型
     */
    func onMusicScorePrepared(pitchModelList: [MusicPitchModel])
    
    /**
     * 打分结束回调
     * - parameter totalScore  总分
     */
    func onMusicScoreFinished(totalScore: Int32)
    
    /**
     * 歌曲的进度
     * - parameter progress  歌曲实时进度
     */
    func onMusicRealTimeProgress(progress: Int)
    
    /**
     * 实时音高回调
     * - parameter pitch      实时音高
     */
    func onMusicRealTimePitch(pitch: Int)
    
    /**
     * 每句歌曲得分
     * - parameter currentScore 分数
     */
    func onMusicSingleScore(currentScore: Int32)
}
